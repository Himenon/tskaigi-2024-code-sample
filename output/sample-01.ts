const mainTask1 = () => {
  const subTask = () => {};
  return subTask();
};
mainTask1();
