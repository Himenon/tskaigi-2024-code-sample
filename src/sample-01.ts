import * as fs from "node:fs";

import * as Printer from "./utils";

export const run = () => {
  const sourceText = `const mainTask = () => {
    const subTask = () => {}
    return subTask();
  }
  mainTask();
  `;

  const code = Printer.generateCodeByPlainText(sourceText);

  fs.writeFileSync("output/sample1.ts", code, "utf8");

  console.log("Generated output/sample1.ts");
};
