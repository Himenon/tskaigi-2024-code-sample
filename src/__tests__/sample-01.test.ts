import * as fs from "node:fs";
import { test, expect } from "vitest";
import * as Module from "../sample-01";

test("Sample1: Code Generate Test", () => {
  Module.run();

  const generateCode = fs.readFileSync("output/sample1.ts", {
    encoding: "utf-8",
  });
  expect(generateCode).toMatchSnapshot();
});
