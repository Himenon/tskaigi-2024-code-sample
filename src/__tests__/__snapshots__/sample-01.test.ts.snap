// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sample1: Code Generate Test 1`] = `
"const mainTask1 = () => {
    const subTask = () => { };
    return subTask();
};
mainTask1();
// SampleCode
"
`;
